presto-orc/src/test/java/com/facebook/presto/orc/OrcTester.java

```diff
@@ -651,15 +705,24 @@ private static void assertFileContentsPresto(
             Format format,
             boolean isHiveWriter,
             boolean useSelectiveOrcReader,
-            List<Map<Integer, Map<Subfield, TupleDomainFilter>>> filters)
-            throws IOException
+            List<Map<Integer, Map<Subfield, TupleDomainFilter>>> filters,
+            Optional<OrderChecker> orderChecker)
+            throws Exception
     {
         OrcPredicate orcPredicate = createOrcPredicate(types, expectedValues, format, isHiveWriter);
         if (useSelectiveOrcReader) {
             assertFileContentsPresto(types, tempFile.getFile(), expectedValues, orcEncoding, orcPredicate, Optional.empty(), ImmutableList.of(), ImmutableMap.of(), ImmutableMap.of());
 
             for (Map<Integer, Map<Subfield, TupleDomainFilter>> columnFilters : filters) {
-                assertFileContentsPresto(types, tempFile.getFile(), filterRows(types, expectedValues, columnFilters), orcEncoding, orcPredicate, Optional.of(columnFilters), ImmutableList.of(), ImmutableMap.of(), ImmutableMap.of());
+                List<List<?>> filteredRows = filterRows(types, expectedValues, columnFilters);
+                if (!orderChecker.isPresent()) {
+                    assertFileContentsPresto(types, tempFile.getFile(), filteredRows, orcEncoding, orcPredicate, Optional.of(columnFilters), ImmutableList.of(), ImmutableMap.of(), ImmutableMap.of());
+                }
+                else {
+                    orderChecker.get().start();
+                    assertFileContentsPresto(types, tempFile.getFile(), filteredRows, orcEncoding, orcPredicate, Optional.of(columnFilters), ImmutableList.of(), ImmutableMap.of(), ImmutableMap.of());
+                    assertEquals(orderChecker.get().end(), true, "Actual order of filter calls doesn't match desired");
```

Comment:
Can you include the expected and actual orders in the error message to help troubleshooting?